opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -1042,8 +1042,8 @@ def start_as_current_span(
             end_on_exit=end_on_exit,
             record_exception=record_exception,
             set_status_on_exception=set_status_on_exception,
-        ) as span_context:
-            yield span_context
+        ):
+            yield span
```

Comment:
Shouldn't this be

```suggestion
        ) as span:
            yield span
```

Reply:
I'm open to this suggestion, but does it make a difference? I ran some tests locally and it doesn't appear to because `span` is already in scope and has the same value as would be assigned by using `as span`, but happy to make the change if I'm missing something or if you think it would be clearer.

Reply:
I see your point, @pmcollins but I think it would be clearer, please change :v: